### PR TITLE
Store: don't marshall/unmarshall the dashboard when building a summary

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
image

marshalling is the most expensive part of building the dashboard summary and we don't really need it for indexing purposes

### CODE REVIEW

File: pkg/services/store/kind/dashboard/summary.go [hunk 1]
```diff
@@ -30,16 +29,6 @@ func GetObjectSummaryBuilder() models.ObjectSummaryBuilder {
 // This implementation moves datasources referenced by internal ID or name to UID
 func NewStaticDashboardSummaryBuilder(lookup DatasourceLookup) models.ObjectSummaryBuilder {
 	return func(ctx context.Context, uid string, body []byte) (*models.ObjectSummary, []byte, error) {
-		var parsed map[string]interface{}
```
Comment (ArturWierzbicki) [inline pkg/services/store/kind/dashboard/summary.go:33]: maybe should we control this with a flag? @ryantxu

**Comment by ArturWierzbicki** [comment]
> I am torn on this -- yes it would improve write performance; but then have less consistent results on read 😬
> 
> With the marshal it will:
> 
> 1. not have redundant (perhaps wrong!) nested UID
> 2. etag will be consistent regardless of formatting

What about splitting this method into two? `GetSummary` and `Sanitize/Normalize/..`

**Comment by ryantxu** [comment]
Ahhh -- I see this impacts the existing search interface, not just storage!  I'll take a look today and propose a better solution

**Review by ryantxu — APPROVED** [review]

**Comment by ArturWierzbicki** [comment]
thanks @ryantxu, works like a dream!! 
image

**Pull request merged**